### DISCUSSION TITLE
How to get all tailwind CSS classes?

### DISCUSSION THREAD

**Discussion by adaboese** [tailwindlabs/tailwindcss#12653] [General]
Is there a constant or something that I could reference to get all possible tailwind CSS classes?

**Comment by wongjn** [comment]
This would not really be possible for v3 – there's an infinite amount of classes possible (due to the arbitrary features) and even without these, if you really wanted every single class possible, you would need to generate every possible combination of the variants, (like `dark:first:hover:focus:md:visited:last:only:…:m-1`) so there would be all the "base" classes plus all these permutations of variants and thus the total number of possible classes would be something like (71![^1] + 1) × number of "base" classes (maths probably not correct here, but something like this). The "real" number would be slightly less since some variants don't make sense together, i.e. `sm:md:lg:xl:m-1` but this obscenely large number should demonstrate the point.

[^1]: 71 factorial, equal to 850478588567862317521167644239926010288584608120796235886430763388588680378079017697280000000000000000

**Reply by adaboese** [reply]
I don't need to know every possible permutation. I just need to know a list of every possible token that makes the entire class, i.e. In your example of `dark:first:hover:focus`, I need `dark`, `first`, `hover`, etc.

**Comment by ony3000** [comment]
You may want to refer to [this document](https://tailwindcss.com/docs/hover-focus-and-other-states#quick-reference).
